web-registry/src/pages/BatchDetails.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from 'react';
+import { Box } from '@mui/material';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import OutlinedButton from 'web-components/lib/components/buttons/OutlinedButton';
+import Section from 'web-components/lib/components/section';
+import Title from 'web-components/lib/components/title';
+import { Loading } from 'web-components/lib/components/loading';
+
+import { getBatchWithSupplyForDenom } from '../lib/ecocredit';
+import { getMetadata } from '../lib/metadata-graph';
+import { useBatchDetailsQuery } from '../generated/graphql';
+
+import type { BatchInfoWithSupply } from '../types/ledger/ecocredit';
+import type { BatchMetadataLD } from '../generated/json-ld';
+import {
+  BatchInfoGrid,
+  BatchMetadata,
+  BatchTotalsGrid,
+} from '../components/molecules';
+import { NotFoundPage } from './NotFound';
+import { useWallet } from '../lib/wallet';
+import { useEcocredits } from '../hooks';
+
+export const BatchDetails: React.FC = () => {
+  const { batchDenom } = useParams();
+  const [loading, setLoading] = useState(true);
+  const [batch, setBatch] = useState<BatchInfoWithSupply>();
+  const [metadata, setMetadata] = useState<BatchMetadataLD>();
+  const navigate = useNavigate();
+  const walletContext = useWallet();
+  const accountAddress = walletContext.wallet?.address;
+  const userEcocredits = useEcocredits(accountAddress);
+  const isUserBatch = userEcocredits.some(c => c.batch_denom === batchDenom);
+
+  useEffect(() => {
+    const fetch = async (): Promise<void> => {
+      if (batchDenom) {
+        try {
+          setLoading(true);
+          const batch = await getBatchWithSupplyForDenom(batchDenom);
+          setBatch(batch);
+          if (batch.metadata) {
+            const { data } = await getMetadata(batch.metadata);
+            setMetadata(data.metadata);
+          }
+        } catch (err) {
+          console.error(err); // eslint-disable-line no-console
+        } finally {
+          setLoading(false);
+        }
+      }
+    };
+    fetch();
+  }, [batchDenom]);
+
+  const { data: offchainData } = useBatchDetailsQuery({
```

Comment:
shouldn't we also pull the loading state from this query to know whether we display <Loading /> or not?

Reply:
Yes, good idea! I threw it in last minute - will update